network/src/commonMain/kotlin/com/wire/kalium/network/api/base/model/UserDTO.kt

```diff
@@ -39,7 +40,8 @@ data class UserDTO(
     @SerialName("qualified_id") val id: UserId,
     @SerialName("service") val service: ServiceDTO?,
     @SerialName("sso_id") val ssoID: UserSsoIdDTO?,
-    @SerialName("team") val teamId: TeamId?
+    @SerialName("team") val teamId: TeamId?,
+    @SerialName("supported_protocols") val supportedProtocols: List<SupportedProtocolDTO> = listOf(SupportedProtocolDTO.PROTEUS)
```

Comment:
I _hope_ we don't forget to remove this hardcoded `PROTEUS` here.

Ideally we should bubble up this to the upper layers to actually have the logic to decide what should be the supported protocols.

I can see the case for leaving it as a default argument, as it would be a lot of repetition and could actually make it harder for us to replace it in multiple places.

Can we, instead of the default parameter, add a secondary constructor that will put the default `PROTEUS` argument, and annotate it with `@Deprecated`?

Other than the warning, at least we will be able to find usages of the deprecated constructor more easily when needed in the future.

``` kotlin
    @SerialName("supported_protocols") val supportedProtocols: List<SupportedProtocolDTO>
) {

    @Deprecated("supportedProtocols will become a mandatory field. Specify the supported protocols.")
    constructor(
        accentId: Int,
        assets: List<UserAssetDTO>,
        deleted: Boolean?,
        email: String?,
        expiresAt: String?,
        handle: String?,
        nonQualifiedId: NonQualifiedUserId,
        name: String,
        locale: String,
        managedByDTO: ManagedByDTO?,
        phone: String?,
        id: UserId,
        service: ServiceDTO?,
        ssoID: UserSsoIdDTO?,
        teamId: TeamId?
    ) : this(
        accentId,
        assets,
        deleted,
        email,
        expiresAt,
        handle,
        nonQualifiedId,
        name,
        locale,
        managedByDTO,
        phone,
        id,
        service,
        ssoID,
        teamId,
        listOf(SupportedProtocolDTO.PROTEUS)
    )
}
```

Reply:
As long as this field exists it will be optional because it's something that a user declare. If a user hasn't set it yet we assume only proteus is supported.

I can make it optional and introduce this assumption at the mapping layer in logic instead. I suspect I need to do it in many places though since we map directly from api to entity and api to model, but I can live it.

Reply:
Moved the default handling to the logic layer. I'll do refactoring PR on develop to clean up code duplication in  UserMapper and PublicUserMapper